src/redis-default.js

```diff
@@ -0,0 +1,104 @@
+import { promisify } from 'util'
+
+class RedisDefaultStore {
+  constructor (client, options = {}) {
+    const invalidClientError = new TypeError(
+      'First parameter must be a valid RedisClient instance.'
+    )
+
+    try {
+      if (client.constructor.name !== 'RedisClient') {
+        throw invalidClientError
+      }
+    } catch (err) {
+      throw invalidClientError
+    }
+    this.client = client
+    this.prefix = options.prefix || 'axios-cache'
+    this.maxScanCount = options.maxScanCount || 1000
+    this.getAsync = promisify(client.get).bind(client)
+    this.psetexAsync = promisify(client.psetex).bind(client)
+    this.delAsync = promisify(client.del).bind(client)
+    this.scanAsync = promisify(client.scan).bind(client)
+  }
+
+  calculateTTL (value) {
+    const now = Date.now()
+
+    if (value.expires && value.expires > now) {
+      return value.expires - now
+    }
+
+    // If there is no expires in value or the provided expire is before the current time
+
+    return -1
+  }
+
+  transformKey (key) {
+    return this.prefix + '_' + key
+  }
+
+  async getItem (key) {
+    const item = (await this.getAsync(this.transformKey(key))) || null
+
+    return JSON.parse(item)
+  }
+
+  async setItem (key, value) {
+    const computedKey = this.transformKey(key)
+
+    const ttl = this.calculateTTL(value)
+
+    if (ttl > 0) {
+      await this.psetexAsync(computedKey, ttl, JSON.stringify(value))
```

Comment:
should we set the entire value along with the `expires` keys?

Reply:
Yes. `axios-cache-adapter` requires the format like that only.